cfme/containers/provider/openshift.py

```diff
@@ -120,18 +192,38 @@ def from_config(cls, prov_config, prov_key, appliance=None):
                 endpoints[endp] = OpenshiftDefaultEndpoint(**prov_config['endpoints'][endp])
             elif HawkularEndpoint.name == endp:
                 endpoints[endp] = HawkularEndpoint(**prov_config['endpoints'][endp])
+            elif AlertsEndpoint.name == endp:
+                endpoints[endp] = AlertsEndpoint(**prov_config['endpoints'][endp])
             # TODO Add Prometheus and logic for having to select or the other based on metrcis_type
             else:
                 raise Exception('Unsupported endpoint type "{}".'.format(endp))
 
+        http_proxy = prov_config.get('settings', {}).get('proxy', {}).get('http_proxy')
+        adv_http = prov_config.get('settings', {}).get('advanced', {}).get('adv_http')
+        adv_https = prov_config.get('settings', {}).get('advanced', {}).get('adv_https')
+        no_proxy = prov_config.get('settings', {}).get('advanced', {}).get('no_proxy')
+        image_repo = prov_config.get('settings', {}).get('advanced', {}).get('image_repo')
+        image_reg = prov_config.get('settings', {}).get('advanced', {}).get('image_reg')
+        image_tag = prov_config.get('settings', {}).get('advanced', {}).get('image_tag')
+        cve_loc = prov_config.get('settings', {}).get('advanced', {}).get('cve_loc')
```

Comment:
__OPTIONAL__
Change all the above 8 lines to:
```python
settings = prov_config.get('settings', {})
advanced = settings.get('advanced', {})
http_proxy = settings.get('proxy', {}).get('http_proxy')
adv_https, adv_https, no_proxy, image_repo, image_reg, image_tag, cve_loc = [
    advanced.get(field) for field in
    ('adv_https', 'adv_https', 'no_proxy', 'image_repo', 'image_reg', 'image_tag', 'cve_loc')
]
```
Note that this is just my approach for lines count reduction but probably it's too complicated and the current is good enough :-) up to you.

Reply:
:+1: This is a great suggestion